image_explorer/image_explorer.py

```diff
@@ -287,16 +286,30 @@ def _inner_content(self, tag):
             return u''.join(html.tostring(e) for e in tag)
         return None
 
-    def _get_description(self, xmltree):
+    def _get_description(self, xmltree, absolute_urls=False):
         """
         Parse the XML to get the description information
         """
         description = xmltree.find('description')
         if description is not None:
-            return self._inner_content(description)
+            description = self._inner_content(description)
+            if absolute_urls:
+                description = self._change_relative_url_to_absolute(description)
```

Comment:
Since all values are needed after getting the inner content value, and since the actual modification happens on the inner content value itself, moving the logic of 
```PYTHON
if absolute_urls:
    variable = self._change_relative_url_to_absolute(value)
```
inside [_inner_content](https://github.com/edx-solutions/xblock-image-explorer/pull/48/files#diff-017f8344b19d5b1f39699e1b83608a8fR281) seems much better since it'll keep the code cleaner, and the logic inside a single module. You only need to pass `absolute_urls` to `self._inner_content(value, absolute_urls=absolute_urls)`

This will affect:
- Line [295](https://github.com/edx-solutions/xblock-image-explorer/pull/48/files#diff-017f8344b19d5b1f39699e1b83608a8fR295)
- Line [327](https://github.com/edx-solutions/xblock-image-explorer/pull/48/files#diff-017f8344b19d5b1f39699e1b83608a8fR327)
- Line [337](https://github.com/edx-solutions/xblock-image-explorer/pull/48/files#diff-017f8344b19d5b1f39699e1b83608a8fR337)

Reply:
There are 2 functions which are mainly using this _inner_context method, Specifically 'student_view' and 'student_view_data' and we only want this functionality for  'student_view_data' (For mobile calls).

Reply:
Yes, this is why we're gonna let the `_inner_content` method to accept the optional `absolute_urls` parameter which defaults to `False`.

Reply:
Okay, that way would create a chain for passing the parameter because there are also few methods in between. (For example: student_view_data calls _get_description which calls _inner_content).

Reply:
It's okay since that won't goo deeper, and not all methods are going to pass it.